fastestimator/trace/metric/bleu_score.py

```diff
@@ -0,0 +1,123 @@
+       smaller denominators; therefore, we give them proportionally smaller smoothed counts. Instead of scaling to 1/(2^k),
+       Chen and Cherry suggests dividing by 1/ln(len(T)), where T is the length of the translation.
+
+
+    Args:
+        true_key: Name of the key that corresponds to ground truth in the batch dictionary.
+        pred_key: Name of the key that corresponds to predicted score in the batch dictionary.
+        mode: What mode(s) to execute this Trace in. For example, "train", "eval", "test", or "infer". To execute
+            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
+            like "!infer" or "!train".
+        ds_id: What dataset id(s) to execute this Trace in. To execute regardless of ds_id, pass None. To execute in all
+            ds_ids except for a particular one, you can pass an argument like "!ds1".
+        output_name: Name of the key to store back to the state.
+        n_gram: Number of grams used to calculate bleu score(default: 4)
+        per_ds: Whether to automatically compute this metric individually for every ds_id it runs on, in addition to
+            computing an aggregate across all ds_ids on which it runs. This is automatically False if `output_name`
+            contains a "|" character.
+
+    """
+    def __init__(self,
+                 true_key: str,
+                 pred_key: str,
+                 mode: Union[None, str, Iterable[str]] = ("eval", "test"),
+                 ds_id: Union[None, str, Iterable[str]] = None,
+                 output_name: str = 'bleu_score',
+                 n_gram: int = 4,
+                 per_ds: bool = True) -> None:
+        super().__init__(inputs=(true_key, pred_key), outputs=output_name, mode=mode, ds_id=ds_id)
+        self.n_gram = n_gram
+        self.weights = self.get_output_weights()
+        self.smoothing_function = SmoothingFunction().method4
+        self.sentence_level_bleu_scores = []
+        self.per_ds = per_ds
+
+    @property
+    def true_key(self) -> str:
+        return self.inputs[0]
+
+    @property
+    def pred_key(self) -> str:
+        return self.inputs[1]
+
+    def get_output_weights(self):
+        if self.n_gram > 0:
+            return (1 / self.n_gram, ) * self.n_gram
+        else:
+            ValueError("N Gram should be an integer greater than")
+
+    def on_epoch_begin(self, data: Data) -> None:
+        self.sentence_level_bleu_scores = []
+
+    def get_sentence_bleu_score(self, input_data):
+        return sentence_bleu(get_formated_reference(input_data[0]),
+                             get_formated_hypotheses(input_data[1]),
+                             weights=self.weights,
+                             smoothing_function=self.smoothing_function)
+
+    def on_batch_end(self, data: Data) -> None:
+        y_pred, y_true = to_number(data['pred']), to_number(data['target_real'])
+        if len(y_pred.shape) > len(y_true.shape):
```

Comment:
if both y_pred and y_true are in one-hot format will the sentence_blue work? or is it expecting both to be argmaxed?